VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionService.java

```diff
@@ -16,183 +16,159 @@
+    Call<CourseDiscussionInfo> getCourseDiscussionInfo(@Path("course_id") String courseId);
 
 
     @GET("/api/discussion/v1/courses/{course_id}/")
-    CourseDiscussionInfo getCourseDiscussionInfoWithCacheEnabled(@Path("course_id") String courseId)
-            throws HttpException;
+    Call<CourseDiscussionInfo> getCourseDiscussionInfoWithCacheEnabled(@Path("course_id") String courseId);
 
 
     @GET("/api/discussion/v1/course_topics/{course_id}")
-    CourseTopics getCourseTopics(@Path("course_id") String courseId) throws HttpException;
+    Call<CourseTopics> getCourseTopics(@Path("course_id") String courseId);
 
 
     @GET("/api/discussion/v1/course_topics/{course_id}")
-    CourseTopics getSpecificCourseTopics(@Path("course_id") String courseId,
-                                         @Query("topic_id") List<String> topicIds)
-            throws HttpException;
-
-
-    @GET("/api/discussion/v1/threads/")
-    Page<DiscussionThread> getThreadList(@Query("course_id") String courseId,
-                                         @Query("topic_id") List<String> topicIds,
-                                         @Query("view") String view,
-                                         @Query("order_by") String orderBy,
-                                         @Query("page_size") int pageSize,
-                                         @Query("page") int page,
-                                         @Query("requested_fields") List<String> requestedFields)
-            throws HttpException;
-
-
-    @GET("/api/discussion/v1/threads/")
-    Page<DiscussionThread> getFollowingThreadList(@Query("course_id") String courseId,
-                                                  @Query("following") String following,
-                                                  @Query("view") String view,
-                                                  @Query("order_by") String orderBy,
-                                                  @Query("page_size") int pageSize,
-                                                  @Query("page") int page,
-                                                  @Query("requested_fields")
-                                                  List<String> requestedFields)
-            throws HttpException;
+    Call<CourseTopics> getSpecificCourseTopics(@Path("course_id") String courseId,
+                                               @Query("topic_id") List<String> topicIds);
 
-    @GET("/api/discussion/v1/threads/")
-    Page<DiscussionThread> searchThreadList(@Query("course_id") String courseId,
-                                            @Query("text_search") String text,
-                                            @Query("page_size") int pageSize,
-                                            @Query("page") int page,
-                                            @Query("requested_fields") List<String> requestedFields)
-            throws HttpException;
+
+    @GET("/api/discussion/v1/threads?" + QUERY_PAGE_SIZE)
```

Comment:
Is this a functional change? This seems like it might be a required parameter for the new retrofit, but does it actually change the functionality at all? What happens if our query is above the page size? How do we get the extra data?